src/main/java/org/opensearch/ad/rest/handler/IndexAnomalyDetectorJobActionHandler.java

```diff
@@ -106,37 +110,76 @@ public IndexAnomalyDetectorJobActionHandler(
         this.xContentRegistry = xContentRegistry;
         this.transportService = transportService;
         this.adTaskManager = adTaskManager;
+        this.recorder = recorder;
     }
 
     /**
      * Start anomaly detector job.
      * 1. If job doesn't exist, create new job.
      * 2. If job exists: a). if job enabled, return error message; b). if job disabled, enable job.
      * @param detector anomaly detector
+     * @param listener Listener to send responses
      */
-    public void startAnomalyDetectorJob(AnomalyDetector detector) {
+    public void startAnomalyDetectorJob(AnomalyDetector detector, ActionListener<AnomalyDetectorJobResponse> listener) {
```

Comment:
Just to clarify: this start listener is created & injected throughout the job handler so that whenever the job response is received, there's the extra step of trying to index results and update detector state with a 60s delay, correct?

Reply:
yes